src/DataValues/DescriptionEncoders/TimeValueDescriptionEncoder.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace SMW\DataValues\DescriptionEncoders;
+
+use SMWTimeValue as TimeValue;
+use SMWDITime as DITime;
+use SMW\Query\Language\ThingDescription;
+use SMW\Query\Language\ValueDescription;
+use SMW\Query\Language\Disjunction;
+use SMW\Query\Language\Conjunction;
+use InvalidArgumentException;
+use DateTime;
+use DateInterval;
+
+/**
+ * @private
+ *
+ * @license GNU GPL v2+
+ * @since 2.3
+ *
+ * @author mwjames
+ */
+class TimeValueDescriptionEncoder extends DescriptionEncoder {
```

Comment:
When I read this name I figured I'd do the opposite: construct a string from a Description object. How about something like DescriptionDeserializer?